homeassistant/components/light/yeelight.py

```diff
@@ -6,157 +6,283 @@
+        self._bulb.get_properties()
+
+    @_cmd
+    def set_brightness(self, brightness, duration):
+        """Set bulb brightness."""
+        if brightness:
+            _LOGGER.debug("Setting brightness: %s", brightness)
+            self._bulb.set_brightness(brightness / 255 * 100,
+                                      duration=duration)
+
+    @_cmd
+    def set_rgb(self, rgb, duration):
+        """Set bulb's color."""
+        if rgb and self.supported_features & SUPPORT_RGB_COLOR:
+            _LOGGER.debug("Setting RGB: %s", rgb)
+            self._bulb.set_rgb(rgb[0], rgb[1], rgb[2], duration=duration)
+
+    @_cmd
+    def set_colortemp(self, colortemp, duration):
+        """Set bulb's color temperature."""
+        if colortemp and self.supported_features & SUPPORT_COLOR_TEMP:
+            temp_in_k = color_temperature_mired_to_kelvin(colortemp)
+            _LOGGER.debug("Setting color temp: %s K", temp_in_k)
+
+            self._bulb.set_color_temp(temp_in_k, duration=duration)
+
+    @_cmd
+    def set_default(self):
+        """Set current options as default."""
+        self._bulb.set_default()
+
+    @_cmd
+    def set_flash(self, flash):
+        """Activate flash."""
+        if flash:
+            from yeelight import RGBTransition, SleepTransition, Flow
+            if self._bulb.last_properties["color_mode"] != 1:
+                _LOGGER.error("Flash supported currently only in RGB mode.")
+                return
+
+            transition = self.config[ATTR_TRANSITION]
+            if flash == FLASH_LONG:
+                count = 1
+                duration = transition * 5
+            if flash == FLASH_SHORT:
+                count = 1
+                duration = transition * 2
+
+            red, green, blue = self.rgb_color
+
+            transitions = list()
+            transitions.append(
+                RGBTransition(255, 0, 0, brightness=10, duration=duration))
+            transitions.append(SleepTransition(
+                duration=transition))
+            transitions.append(
+                RGBTransition(red, green, blue, brightness=self.brightness,
+                              duration=duration))
+
+            # from pprint import pformat as pf
+            # _LOGGER.error(pf(transitions))
+
+            flow = Flow(count=count, transitions=transitions)
+            self._bulb.start_flow(flow)
 
-        if ATTR_BRIGHTNESS in kwargs:
-            bright = int(kwargs[ATTR_BRIGHTNESS] * 100 / 255)
-            self._bulb.set_brightness(bright)
-            self._bright = kwargs[ATTR_BRIGHTNESS]
+    def turn_on(self, **kwargs):
+        """Turn the bulb on."""
+        brightness = kwargs.get(ATTR_BRIGHTNESS)
+        colortemp = kwargs.get(ATTR_COLOR_TEMP)
+        rgb = kwargs.get(ATTR_RGB_COLOR)
+        flash = kwargs.get(ATTR_FLASH)
+
+        # white bulb has problems with duration > 9000, doesn't always start
+        transition = self.config["transition"]
```

Comment:
Should be `self.config[CONF_TRANSITION]`. Also, the turn_on transition argument is in seconds, not ms.